src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -617,6 +633,230 @@ BoundExpression convertArrayElement(BoundExpression element, TypeSymbol elementT
             return result.Update(result.Placeholder, result.CollectionCreation, result.Elements, targetType);
         }
 
+        private BoundCollectionLiteralExpression BindDictionaryCollectionLiteral(
+            BoundUnconvertedCollectionLiteralExpression node,
+            NamedTypeSymbol targetType,
+            bool wasCompilerGenerated,
+            BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(targetType.IsInterface
+                || targetType.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_Dictionary_KV), TypeCompareKind.AllIgnoreOptions));
+            Debug.Assert(targetType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics.Length == 2);
+
+            var syntax = node.Syntax;
+            var collectionType = targetType.IsInterface
+                ? GetWellKnownType(WellKnownType.System_Collections_Generic_Dictionary_KV, diagnostics, syntax).Construct(targetType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics)
+                : targetType;
+
+            var analyzedArguments = AnalyzedArguments.GetInstance();
+            // PROTOTYPE: Should we use Dictionary<TKey, TValue>(int capacity) constructor when the size is known?
+            var collectionCreation = BindClassCreationExpression(syntax, collectionType.Name, syntax, collectionType, analyzedArguments, diagnostics);
+            collectionCreation.WasCompilerGenerated = true;
+            analyzedArguments.Free();
+
+            ImmutableArray<BoundExpression> elements;
+            var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, collectionType) { WasCompilerGenerated = true };
+
+            if (node.Elements.Length == 0)
+            {
+                elements = ImmutableArray<BoundExpression>.Empty;
+            }
+            else
+            {
+                // PROTOTYPE: Test with missing indexer, multiple indexers, readonly indexer, etc.
+                var lookupResult = LookupResult.GetInstance();
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                // PROTOTYPE: Should we use the well-known Dictionary<TKey, TValue>.this[TKey] { get; set; }
+                // member rather than relying on lookup?
```

Comment:
technically no.  teh spec defines everything as a translation.  so lookup is appropriate.  That said, we also assume the BCL is well-behaved.  So not finding the normal indexer would be super weird.

Reply:
also, the spec says that every element is its own translation (also applies to lists).  So you could bind to a different .Add method each time.  We should either prototype this, or update this to match spec, or update spec to match this.